Frameworks/CoreGraphics/CGColor.mm

```diff
@@ -56,31 +173,39 @@ CGColorRef CGColorGetConstantColor(CFStringRef name) {
 */
 void CGColorRelease(CGColorRef color) {
     RETURN_IF(!color);
-    CFRelease(color);
+    CFRelease(static_cast<CFTypeRef>(color));
 }
 
 /**
  @Status Interoperable
 */
 CGColorRef CGColorRetain(CGColorRef color) {
-    CFRetain((id)color);
-
+    RETURN_NULL_IF(!color);
+    CFRetain(static_cast<CFTypeRef>(color));
     return color;
 }
 
 /**
  @Status Caveat
- @Notes only supports RGB
+ @Notes only supports RGB & Monochrome
 */
 CGColorRef CGColorCreate(CGColorSpaceRef colorSpace, const CGFloat* components) {
     RETURN_NULL_IF(!colorSpace);
     RETURN_NULL_IF(!components);
-    CGColorRef ret = static_cast<CGColorRef>(
-        [[__LazyUIColor colorWithRed:components[0] green:components[1] blue:components[2] alpha:components[3]] retain]);
-    [static_cast<UIColor*>(ret) setColorSpace:colorSpace];
-    return ret;
-}
+    RETURN_NULL_IF(CGColorSpaceGetNumberOfComponents(colorSpace) == 0);
```

Comment:
kCGColorSpaceModelUnknown  would do.

---
In reply to: [113351799](https://github.com/Microsoft/WinObjC/pull/2538#discussion_r113351799) [](ancestors = 113351799)